src/app/routes/api/v3/admin/forms/__tests__/admin-forms.presign.routes.spec.ts

```diff
@@ -0,0 +1,528 @@
+          fileType: VALID_UPLOAD_FILE_TYPES[1],
+        })
+
+      // Assert
+      expect(response.status).toEqual(400)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: {
+            key: 'fileId',
+            message: '"fileId" is not allowed to be empty',
+          },
+        }),
+      )
+    })
+
+    it('should return 400 when body.fileType is missing', async () => {
+      // Act
+      const response = await request
+        .post(`/admin/forms/${new ObjectId()}/logos/presign`)
+        .send({
+          fileId: 'some id',
+          fileMd5Hash: SparkMD5.hash('test file name'),
+          // Missing fileType.
+        })
+
+      // Assert
+      expect(response.status).toEqual(400)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: { key: 'fileType' },
+        }),
+      )
+    })
+
+    it('should return 400 when body.fileType is invalid', async () => {
+      // Act
+      const response = await request
+        .post(`/admin/forms/${new ObjectId()}/logos/presign`)
+        .send({
+          fileId: 'some id',
+          fileMd5Hash: SparkMD5.hash('test file name'),
+          fileType: 'some random type',
+        })
+
+      // Assert
+      expect(response.status).toEqual(400)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: {
+            key: 'fileType',
+            message: `"fileType" must be one of [${VALID_UPLOAD_FILE_TYPES.join(
+              ', ',
+            )}]`,
+          },
+        }),
+      )
+    })
+
+    it('should return 400 when body.fileMd5Hash is missing', async () => {
+      // Act
+      const response = await request
+        .post(`/admin/forms/${new ObjectId()}/logos/presign`)
+        .send({
+          fileId: 'some id',
+          // Missing fileMd5Hash
+          fileType: VALID_UPLOAD_FILE_TYPES[2],
+        })
+
+      // Assert
+      expect(response.status).toEqual(400)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: { key: 'fileMd5Hash' },
+        }),
+      )
+    })
+
+    it('should return 400 when body.fileMd5Hash is not a base64 string', async () => {
+      // Act
+      const response = await request
+        .post(`/admin/forms/${new ObjectId()}/logos/presign`)
+        .send({
+          fileId: 'some id',
+          fileMd5Hash: 'rubbish hash',
+          fileType: VALID_UPLOAD_FILE_TYPES[2],
+        })
+
+      // Assert
+      expect(response.status).toEqual(400)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: {
+            key: 'fileMd5Hash',
+            message: '"fileMd5Hash" must be a valid base64 string',
+          },
+        }),
+      )
+    })
+
+    it('should return 400 when creating presigned POST URL object errors', async () => {
+      // Arrange
+      // Mock error.
+      jest
+        .spyOn(aws.s3, 'createPresignedPost')
+        // @ts-ignore
```

Comment:
nit: can remove this comment now that the lint rule is disabled file wide :P

Reply:
sorry got confused with the linter ignore instead of ts-ignore, pls ignore this!!